src/popup.cpp

```diff
@@ -103,8 +103,13 @@ std::vector<std::vector<std::string>> query_popup::fold_query(
     int query_cnt = 0;
     int query_width = 0;
     for( const query_popup::query_option &opt : options ) {
-        const std::string &name = ctxt.get_action_name( opt.action );
-        const std::string &desc = ctxt.get_desc( opt.action, name, opt.filter );
+        // TODO: This function is only called once (on init) so we're not able
+        //   to properly color the key (colour tagging it prevents it from
+        //   being highlighted when selected)... This isn't a too much of
+        //   problem with the default settings because the key is separated
+        //   by parens but still...
+        const std::string &desc = ctxt.get_hint( opt.action, keybinding_hint_state::NONE_AT_ALL,
+                                  opt.filter );
```

Comment:
it's a little hacky but my solution here works https://github.com/CleverRaven/Cataclysm-DDA/pull/64478/commits/cdadc091f83a81f5d886ec7e31d3d06b68b4d80b

Reply:
I refactored the code so it's a little less hacky now.

I couldn't quite get the other solution to work cleanly (although I understand it's possible). Do you think this can be good enough for now and I can come back to this in another PR? Or is this unmergeable in this state you think?

Reply:
If you add a comment to the replacement function to warn that it could break in some edge cases I think it's good enough for now.